### PR TITLE
add cairo_surface_create_for_rectangle()

### PR DESCRIPTION
**Pull request by wrigjl**
add interface for cairo_surface_create_for_rectangle (creates subsurface of an existing surface).

### CODE REVIEW

**Comment by EPashkin** [comment]
LGFM except that it need `cargo fmt`

**Comment by EPashkin** [comment]
👍

**Comment by GuillaumeGomez** [comment]
Looks good to me as well, but please run `cargo fmt`. :)

**Comment by wrigjl** [comment]
already done

**Comment by sdroege** [comment]
:+1:

**Comment by GuillaumeGomez** [comment]
@wrigjl CI doesn't agree. ;) Maybe you forgot to push?

**Comment by wrigjl** [comment]
I didn't forget to push, it looks like it tested the version before I fixed.

https://github.com/gtk-rs/cairo/pull/301/commits/85cd3c03ef43a2478d24cda06b61c9911c7e4338

File: cairo-sys-rs/src/lib.rs [hunk 1]
```diff
@@ -958,6 +958,13 @@ extern "C" {
         width: c_int,
         height: c_int,
     ) -> *mut cairo_surface_t;
+    pub fn cairo_surface_create_for_rectangle(
+        surface: *mut cairo_surface_t,
+	x: c_double,
```
Comment (GuillaumeGomez) [inline cairo-sys-rs/src/lib.rs]: No, look here. You forgot to format this one. ;)

**Comment by wrigjl** [comment]
damn. fixed.

**Comment by GuillaumeGomez** [comment]
Let's wait for the CI now...

**Comment by wrigjl** [comment]
I don't think this diff caused the macos failures.

**Comment by sdroege** [comment]
Yes, unrelated. Build taking too long, we observed that elsewhere too already.

@GuillaumeGomez good to merge :)

**Comment by GuillaumeGomez** [comment]
Thanks!

**Pull request merged**